.mergify.yml

```diff
@@ -0,0 +1,181 @@
+  - name: "label needs: work when Jenkins CI failed - any of the pipeline"
+    conditions:
+      - base~=feature-mergify
+      # Jenkins CI failing - any of the pipeline
+      - status-failure~=^jenkins-ci
+    actions:
+      label:
+        add: 
+          - "needs: work"
+        remove: 
+          - "needs: review"
+
+  # From needs: review to needs: CI
+  - name: "label needs: CI when at least one reviewers approval"
+    conditions:
+      - base~=feature-mergify
+      # Labels
+      - "label!=needs: work"
+      - "label!=needs: preceding PR"
+
+      # Reviewers
+      - "#changes-requested-reviews-by=0"
+      - "#approved-reviews-by>=1"
+
+      # CI green policy, at least Travis should be green
+      - status-success~=continuous-integration/travis-ci/pr
+      # new CI needs to be done
+      - status-neutral~=continuous-integration/jenkins/pr-head
+    actions:
+      label:
+        add: 
+          - "needs: CI"
+        remove: 
+          - "needs: review"
+
+  # Conflict in the PR - needs: work and a comment to notify a user
+  - name: "label needs: work when there is a conflict"
+    conditions:
+      - base~=feature-mergify
+      - conflict
+    actions:
+      label:
+        add:
+          - "needs: work"
+        remove: 
+          - "needs: review"
+      comment:
+        message: This PR cannot be merged due to conflicts. Please rebase to resolve them.
+
+  - name: "add label feature branch for feature branch additions"
+    conditions:
+      - base~=feature-mergify
+      - base~=^feature
+    actions:
+      label:
+        add:
+          - "release-version: feature branch"
+
+  # Ready for integration. Not yet auto merge, will be enabled once carefuly tested
+  - name: label "ready for merge" when ready
+    conditions:
+      - base~=feature-mergify
+      # Labels
+      - label!=do not merge
+      - "label=needs: CI"
+
+      # Reviewers
+      - "#approved-reviews-by>=1"
+      - "#changes-requested-reviews-by=0"
+
+      # CI green policy
+      - status-success~=continuous-integration/travis-ci/pr
+      # Internal Jenkins - we rely on PR head to provide status
+      - status-success~=continuous-integration/jenkins/pr-head
+      # any of the jenkins pipeline needs to be green
+      - status-success~=^jenkins-ci
+    actions:
+      label:
+        add: 
+          - ready for merge
+        remove: 
+          - "needs: CI"
+
+  # Clean-up after merge
+  - name: remove ready for merge when merged
+    conditions:
+      - base~=feature-mergify
+      - merged
+      - label=ready for merge
+    actions:
+      label:
+        remove:
+          - ready for merge
+
+  - name: add "do not merge" label when WIP is in title
+    conditions:
+       - base~=feature-mergify
+       - title~=^(\[wip\]( |:) |\[WIP\]( |:) |wip( |:) |WIP( |:)).*
+    actions:
+      label:
+        add:
+          - do not merge
+
+  # Check if version label is applied
+  - name: release version is a must for merged PRs
```

Comment:
This will require much more thought in the new release scheme. It is envisaged going forward that prior to actual release another bot will add one of these labels depending on what pr type is specified by the author:
'release-version: patch'
'release-version: minor'
'release-version: major'
Then when we are ready to make the release, the release script will determine the actual semantic version of the release based on the content ....

Reply:
I can remove it now or once the release bot is running. This would catch errors, should not cause conflict with the release bot.